src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java

```diff
@@ -579,6 +570,10 @@ public synchronized void notifyGraphResized(int width, int height) {
             this.graphWidth = width;
             this.graphHeight = height;
         }
+
+        public void setPopup(RegionPopUp popup) {
```

Comment:
I think we're going to want to have all the popups updating. You are probably going to need a list of popups and will need to `setSnapshot` and `repaint` all of them. You will also need to be sure to remove the popup from this list when it is closed.